crates/matrix-sdk/src/widget_api/handler/driver.rs

```diff
@@ -36,8 +34,8 @@ impl<'t> From<&'t OpenIDState> for openid::State {
 impl From<OpenIDResult> for openid::State {
     fn from(result: OpenIDResult) -> Self {
         match result {
-            Some(response) => openid::State::Allowed(response),
-            None => openid::State::Blocked,
+            Ok(response) => openid::State::Allowed(response),
+            Err(_) => openid::State::Blocked,
```

Comment:
I updated it to at least print the error and the context of the error so we know why the oidc token state is blocked. Did we fail to get the token because of an error or because the user did not grant permission.

Reply:
Explained in the 1:1 🙂 Can take it over. Short summary: seems like there was no need to use `Error` there (the original implementation simply used `Option`). Printing the error is also not useful, because it's a library code, not the application code.